src/main/java/de/tum/in/www1/artemis/web/rest/settings/ide/IdeSettingsResource.java

```diff
@@ -0,0 +1,145 @@
+package de.tum.in.www1.artemis.web.rest.settings.ide;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.domain.settings.ide.Ide;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMapping;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMappingId;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.settings.IdeRepository;
+import de.tum.in.www1.artemis.repository.settings.UserIdeMappingRepository;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+
+/**
+ * REST controller for managing Ide Settings (IdeSettings).
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class IdeSettingsResource {
+
+    private static final Logger log = LoggerFactory.getLogger(IdeSettingsResource.class);
+
+    private final UserIdeMappingRepository userIdeMappingRepository;
+
+    private final IdeRepository ideRepository;
+
+    private final UserRepository userRepository;
+
+    public IdeSettingsResource(UserIdeMappingRepository userIdeMappingRepository, IdeRepository ideRepository, UserRepository userRepository) {
+        this.userIdeMappingRepository = userIdeMappingRepository;
+        this.ideRepository = ideRepository;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * GET ide-settings: get the predefined IDEs
+     *
+     * @return the ide preferences of the user as IdeMappingDTO
+     */
+    @GetMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<IdeMappingDTO[]> getIdesOfUser() {
+        User user = userRepository.getUser();
+        log.debug("REST request to get IDEs of user {}", user.getLogin());
+
+        var ideMappings = userIdeMappingRepository.findAllByUserId(user.getId());
+        IdeMappingDTO[] ideRecords = ideMappings.stream().map(x -> new IdeMappingDTO(x.getProgrammingLanguage(), x.getIde())).toArray(IdeMappingDTO[]::new);
```

Comment:
We tend to use descriptive variable names.
```suggestion
        IdeMappingDTO[] ideRecords = ideMappings.stream().map(ide -> new IdeMappingDTO(ide.getProgrammingLanguage(), ide.getIde())).toArray(IdeMappingDTO[]::new);
```